contracts/system_tests/TestNFTRewardsFlow.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import './helpers/console.sol';
+import './helpers/TestBaseWorkflow.sol';
+import '../structs/JBTokenAmount.sol';
+import '../libraries/JBCurrencies.sol';
+
+interface Vm {
+  function warp(uint256 x) external;
+}
+
+/// @notice This file tests JBToken related flows
+contract TestNFTRewardsFlow is TestBaseWorkflow {
+  JBController private _controller;
+  JBTokenStore private _tokenStore;
+  JBToken721Store private _nftStore;
+
+  JBProjectMetadata private _projectMetadata;
+  JBFundingCycleData private _fundingCycleData;
+  JBFundingCycleMetadata private _fundingCycleMetadata;
+  JBGroupedSplits[] private _groupedSplits; // Default empty
+  JBFundAccessConstraints[] private _fundAccessConstraints; // Default empty
+  IJBPaymentTerminal[] private _terminals; // Default empty
+
+  uint256 private _projectId = 1;
+  address private _projectOwner;
+  uint256 private _reservedRate = 5000;
+
+  JBNFTRewardDataSourceDelegate private _jbNFTRewardDataSourceDelegate;
+  IJBToken721 private _nft;
+
+  function setUp() public override {
+    super.setUp();
+
+    Vm vm = Vm(address(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D));
```

Comment:
same, you can already use it via `evm.`

Reply:
(I should add the forge-std to follow cheat code addition tho, instead of the hardcoded interface rn)